metrics/character/character.py

```diff
@@ -0,0 +1,123 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""An implementation for calculating CharacTER, a character-based TER variant, useful for machine translation tasks."""
+import cer
+import datasets
+from datasets import Sequence, Value
+
+import evaluate
+
+
+_CITATION = """\
+@inproceedings{wang-etal-2016-character,
+    title = "{C}harac{T}er: Translation Edit Rate on Character Level",
+    author = "Wang, Weiyue  and
+      Peter, Jan-Thorsten  and
+      Rosendahl, Hendrik  and
+      Ney, Hermann",
+    booktitle = "Proceedings of the First Conference on Machine Translation: Volume 2, Shared Task Papers",
+    month = aug,
+    year = "2016",
+    address = "Berlin, Germany",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/W16-2342",
+    doi = "10.18653/v1/W16-2342",
+    pages = "505--510",
+}
+"""
+
+_DESCRIPTION = """\
+CharacTer is a novel character level metric inspired by the commonly applied translation edit rate (TER). It is
+defined as the minimum number of character edits required to adjust a hypothesis, until it completely matches the
+reference, normalized by the length of the hypothesis sentence. CharacTer calculates the character level edit
+distance while performing the shift edit on word level. Unlike the strict matching criterion in TER, a hypothesis
+word is considered to match a reference word and could be shifted, if the edit distance between them is below a
+threshold value. The Levenshtein distance between the reference and the shifted hypothesis sequence is computed on the
+character level. In addition, the lengths of hypothesis sequences instead of reference sequences are used for
+normalizing the edit distance, which effectively counters the issue that shorter translations normally achieve lower
+TER."""
+
+_KWARGS_DESCRIPTION = """
+Calculates how good are predictions given some references, using certain scores
+Args:
+    predictions: a single prediction or a list of predictions to score. Each prediction should be a string with
+     tokens separated by spaces.
+    references: a single reference or a list of reference for each prediction. Each reference should be a string with
+     tokens separated by spaces.
+Returns:
+    count: how many parallel sentences were processed,
+    mean: the mean CharacTER score,
+    median: the median score,
+    std: standard deviation of the score,
+    min: smallest score,
+    max: largest score
+Examples:
+    >>> character = evaluate.load("character")
+    >>> preds = ["this week the saudis denied information published in the new york times",
+                "this is in fact an estimate"]
+    >>> refs = ["saudi arabia denied this week information published in the american new york times",
+                "this is actually an estimate"]
+    >>> results = character.compute(references=refs, predictions=preds)
+    >>> print(results)
+    {
+        'count': 2,
```

Comment:
Question: is there a way to return the per-example / unaggregated score for each example as well?